server/pages/index.js

```diff
@@ -370,11 +371,18 @@ module.exports = function() {
 			}
 		} );
 
-	app.get( '/browsehappy', function( request, response ) {
-		response.status( 200 ).render( 'browsehappy.jade', {
-			urls: generateStaticUrls( request )
+	app.get( '/browsehappy', setUpRoute, function( req, res, next ) {
+		const wpcomRe = /^https?:\/\/[A-z_-]+\.wordpress\.com$/;
+		const primaryBlogUrl = get( req, 'context.user.primary_blog_url', '' );
+		const isWpcom = wpcomRe.test( primaryBlogUrl );
```

Comment:
Yeah, as opposed to Jetpack. My main hesitation was centered around the fact that:
- a JP site may not sit at the root of a domain
- a JP site's "site address" may differ from its "WordPress address"

… and I don't have a non-root Jetpack site that I could set as primary to properly test this. If you do have one and could do a quick test to make sure that `primary_blog_url` is robust enough for those JP sites, I'd be grateful, but otherwise I think that refinement may come in another PR. What do you think?